File: Algorithm/Data_Structure/Bit_Operations/231_easy.go
package Bit_Operations

/*
    @Author  : https://github.com/hakusai22
    @Time    : 2024/10/01 23:41
    @题目     :
    @参考     :
    @时间复杂度:
    @空间复杂度:

 数据范围:

*/

func isPowerOfTwo(n int) bool {
	return n > 0 && n&(n-1) == 0
}

File: Algorithm/Data_Structure/Bit_Operations/231_easy.ts
/**
 * @Time 2024/10/01 23:42
 * @Author  : https://github.com/hakusai22
 * @题目     : https://leetcode.cn/problems/power-of-two/description/
 * @参考     :
 * @时间复杂度:
 * @空间复杂度:
 */

function isPowerOfTwo(n: number): boolean {
    return n > 0 && (n & (n - 1)) === 0;
};